File: src/app/Class/Concert.ts
import { City } from './City';


export class Concert {
  id: number;
  startTime: Date;
  concertName: string;
  concertCity: City;

  constructor(obj?: any) {
    this.id = (obj && obj.id) || 0;
    this.startTime = (obj && obj.startTime) || null;
    this.concertName = (obj && obj.concertName) || "";
    this.concertCity = (obj && obj.concertCity) || null;
  }

  // Getters

  getId(): number {
    return this.id;
  }

  getStartTime(): Date {
    return this.startTime;
  }

  getConcertName(): string {
    return this.concertName;
  }

  getConcertCity(): City {
    return this.concertCity;
  }

  // Setters

  setStartTime(startTime: Date) {
    this.startTime = startTime;
  }

  setConcertName(concertName: string) {
    this.concertName = concertName;
  }

  setConcertCity(concertCity: City) {
    this.concertCity = concertCity;
  }
}

File: src/main/java/musiccube/entities/Concert.java
package musiccube.entities;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "concert")
public class Concert {
    @Id
    @GeneratedValue
    @Column(name = "concertid")
    private int id;

    @Column
    private Date startTime;

    @Column
    private String concertName;

    @ManyToOne
    @JoinColumn(referencedColumnName = "cityid")
    private City concertCity;


    public Concert() {}

    public Concert(Date startTime, String concertName, City concertCity) {
        this.startTime = startTime;
        this.concertName = concertName;
        this.concertCity = concertCity;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Date getStartTime() {
        return startTime;
    }

    public void setStartTime(Date startTime) {
        this.startTime = startTime;
    }

    public String getConcertName() {
        return concertName;
    }

    public void setConcertName(String concertName) {
        this.concertName = concertName;
    }

    public City getConcertCity() {
        return concertCity;
    }

    public void setConcertCity(City concertCity) {
        this.concertCity = concertCity;
    }
}
